### ISSUE TITLE
Transparency issue with stickers

### ISSUE THREAD

**Issue by zach-fuller** [Snapchat/creative-kit-sample#7]
I've noticed that the stickers don't support transparency and have a white background behind any area that would be transparent. Is this the intended functionality, or am I missing something? This occurs even with the sample code.
How would I go about removing this white background?

![IMG_1052](https://user-images.githubusercontent.com/16906332/59746704-1127eb80-923d-11e9-9a62-12dd459526f2.PNG)

**Comment by kasyrm** [comment]
I run into the same problem using SnapSDK 1.3.1. 
Downgrading pod to 1.2.0 fixed the issue for now.

**Comment by cybersamx** [comment]
@zach-fuller This is indeed an issue on Creative Kit and we are working on a **hotfix** right now. The next version `1.3.2` should fix the issue.

There are 2 interim workarounds to preserve the transparency in PNG:

1. Do what @kasyrm suggested. That is to downgrade Creative Kit to version 1.2.0 for now.
1. Put the sticker on the Cloud and pull it from the URL. See the following code:
 
```swift
let snapContent = SCSDKNoSnapContent()
guard let url = URL(string: "https://upload.wikimedia.org/wikipedia/commons/thumb/8/8d/Badge-150755_640.png/240px-Badge-150755_640.png") else {
  return
}
let urlSticker = SCSDKSnapSticker(stickerUrl: url, isAnimated: false)
```

**Comment by cybersamx** [comment]
@zach-fuller @kasyrm I just released Snap Kit `1.3.2`, which fixes this issue. Please run `pod update` to update to the latest version. Thanks.

**Comment by penso** [comment]
@cybersamx this bug reappeared in the latest 1.4.1, sadly.

**Comment by connorford2** [comment]
bump ^

**Comment by penso** [comment]
I fixed this by generating a PNG file and storing it locally, then passing it as a parameter. `snapchatStickerView` is a `UIView` which I want to pass as a sticker.

```swift
class SnapchatShareViewController: UIViewController {
    @IBOutlet weak var snapchatStickerView: UIView!

    func shareSnapchat() {
        #if !(targetEnvironment(simulator))
        let snap = SCSDKNoSnapContent()

        /* Stickers to be used in Snap */
        if let stickerImage = asImage() {
            // There is a bug with transparency, so I have to use a local PNG file see
            // https://github.com/Snapchat/creative-kit-sample/issues/7#issuecomment-599752577
            // snap.sticker = SCSDKSnapSticker(stickerImage: stickerImage)

            if let data = stickerImage.pngData() {
                let filename = getDocumentsDirectory().appendingPathComponent("snapchat.png")
                try? data.write(to: filename)
                snap.sticker = SCSDKSnapSticker(stickerUrl: filename, isAnimated: false)
            }
        }

        snapAPI.startSending(snap) { (error: Error?) in
        }
        #endif
    }
    private func asImage() -> UIImage? {
        guard let view = snapchatStickerView else { return nil }
        UIGraphicsBeginImageContextWithOptions(view.bounds.size, view.isOpaque, UIScreen.main.scale)
        view.drawHierarchy(in: view.bounds, afterScreenUpdates: true)
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return UIImage(cgImage: (image?.cgImage)!)
    }

    private func getDocumentsDirectory() -> URL {
        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
        return paths[0]
    }
```
}

**Comment by cybersamx** [comment]
@penso @connorford2 I am not entirely sure what happened with Snap Kit 1.4. And @penso thanks for the workaround to this issue. 

Regretfully I no longer have access to the Snap Kit SDK after leaving Snap. Pinging @eberglundsc @tphillipssc @davidxiasc @bkirbysc who can help you on address the issue of full transparent sticker support.

**Comment by bkirbysc** [comment]
This was fixed in 1.4.2 https://snapkit.com/docs/changelog/

**Comment by cybersamx** [comment]
👍
